File: treatment/src/main/java/com/hartwig/actin/treatment/database/config/CohortDefinitionConfig.java
package com.hartwig.actin.treatment.database.config;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class CohortDefinitionConfig implements TrialConfig {

    @NotNull
    @Override
    public abstract String trialId();

    @NotNull
    public abstract String cohortId();

    public abstract boolean open();

    @NotNull
    public abstract String description();
}

File: trial/src/main/kotlin/com/hartwig/actin/trial/config/CohortDefinitionConfig.kt
package com.hartwig.actin.trial.config

data class CohortDefinitionConfig(
    override val nctId: String,
    val cohortId: String,
    val externalCohortIds: Set<String>,
    val evaluable: Boolean,
    val open: Boolean?,
    val slotsAvailable: Boolean?,
    val ignore: Boolean,
    val description: String
) : TrialConfig
